core/websocket.ml

```diff
@@ -203,19 +203,30 @@ module IO(IO: Cohttp.S.IO) = struct
        EndianBytes.BigEndian.set_int64 scratch 0 Int64.(of_int n);
        Buffer.add_subbytes buf scratch 0 8;
     end;
-    if masked then begin
+    begin match masked with
+    | None -> ()
+    | Some random_string ->
       let mask = random_string 4 in
       Buffer.add_string buf mask;
       if len > 0 then xor mask content;
     end;
     Buffer.add_bytes buf content
 
-  let close_with_code buf random_string masked oc code =
+  let close_with_code ?masked buf oc code =
     Buffer.clear buf;
-    write_frame_to_buf ?random_string ~masked buf @@ Frame.close code;
+    write_frame_to_buf ?masked buf @@ Frame.close code;
     write oc @@ Buffer.contents buf
 
-  let make_read_frame ?(buf=Buffer.create 128) ?random_string ~masked ic oc = fun () ->
+  let make_read_frame
+      ?(buf=Buffer.create 128)
+      ~random_string
+      ~masked ic oc = fun () ->
```

Comment:
`masked` should have the same type as `write_frame_to_buf`, `random_string` can be removed then.